### PR TITLE
refactor: turn into dual node package with additional browser bundle

### PR DESCRIPTION
**Pull request by axe312ger**
Requires https://github.com/contentful/contentful-sdk-core/pull/438

* [x] remove node (minified) bundle
* [x] make available in cjs (as bundle for node without ESM support)
* [x] make available in esm
* [x] support ESM only dependencies
* [x] support browser plain + minified
* [x] switch from jest to vitest
* [x] switch to rollup from webpack

# Will fix:

* https://github.com/contentful/contentful.js/issues/2131
* https://github.com/contentful/contentful.js/issues/2122
* https://github.com/contentful/contentful.js/pull/2130
* https://github.com/contentful/contentful.js/issues/1873
* https://github.com/contentful/contentful.js/issues/2214

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      presets: [
+        [
+          '@babel/preset-env',
+          // Please note: This is set to Node.js v8 in order to satisfy ECMA2017 requirements
+          // However, we cannot ensure it will operate without issues on Node.js v8
+          { targets: { node: 8 }, modules: false, bugfixes: true },
+        ],
+      ],
+    }),
+    alias({
```
Comment (axe312ger) [inline rollup.config.js:81]: maybe obsolute cus latest axios defines browser field in package.jon

**Comment by axe312ger** [comment]
Green tests 😱✅

Thank you @mgoudy91 for publishing the [updated esm compatible SDK-CORE](https://github.com/contentful/contentful-sdk-core/pull/438) :)

File: test/output-integration/browser/jest.config.js
Comment (mgoudy91) [inline test/output-integration/browser/jest.config.js:1]: is this supposed to be here?

**Review by mgoudy91 — APPROVED** [review]
This looks great to me!

- Looks like a few places with some commented out code / configuration that would be nice to clean up before merge.
- Just because I'm always afraid of semantic-release: will this publish as a new major?

**Comment by mgoudy91** [comment]
Looks like there's some conflicts as well, I started taking a look but I think it's going to take more time for me to gain context than I thought. Let me know if you'd like help with that and I can take a look

File: test/output-integration/browser/jest.config.js
Comment (axe312ger) [reply test/output-integration/browser/jest.config.js:1]: its gone :)

**Review by mgoudy91 — DISMISSED** [review]

**Review by jjolton-contentful — DISMISSED** [review]

**Pull request merged**